### PR TITLE
upgrade jackson to 2.11 and disable default data typing (1.11.x)

### PR DESCRIPTION
**Pull request by claudio4j**
https://issues.redhat.com/browse/ENTESB-14584

There was an upgrade to resteasy 4.5 that requires jackson 2.11 #8921
If there is any need to allow classes to be deserialized using Default Typing, it should be activated with Default Typing and setting a custom PolymorphicTypeValidator.

### CODE REVIEW

File: install/operator/deploy/crds/syndesis.io_syndesis_crd.yaml [hunk 1]
```diff
@@ -513,8 +513,6 @@ spec:
                           volumeCapacity:
                             type: string
                         type: object
-                      rules:
```
Comment (squakez) [inline install/operator/deploy/crds/syndesis.io_syndesis_crd.yaml:516]: Is this change required by this PR scope?

**Review by squakez — APPROVED** [review]
Just one comment added for your consideration.

File: install/operator/deploy/crds/syndesis.io_syndesis_crd.yaml (hunk 1 above)
Comment (claudio4j) [reply install/operator/deploy/crds/syndesis.io_syndesis_crd.yaml:516]: This CRD file is automatically generated by operator-sdk when the `install/operator/build.sh` runs. The property was removed in this PR https://github.com/syndesisio/syndesis/pull/8979 but was not pushed. This should be pushed anyway.

**Comment by gashcrumb** [comment]
retest this please

**Comment by claudio4j** [comment]
retest this please

**Comment by avano** [comment]
retest this please

**Comment by avano** [comment]
retest this please

**Comment by avano** [comment]
retest this please
